sdks/java/core/src/main/java/org/apache/beam/sdk/util/MergingActiveWindowSet.java

```diff
@@ -448,6 +448,11 @@ public W mergedWriteStateAddress(Collection<W> toBeMerged, W mergeResult) {
     return mergeResult;
   }
 
+  @Override
+  public boolean hasState() {
+    return !activeWindowToStateAddressWindows.isEmpty();
```

Comment:
Intent is to ask if there are any currently active windows, ie what would be saved when the bundle is committed.